File: backend/src/main/java/net/zorphy/backend/site/core/ws/dto/GameRoom.java
package net.zorphy.backend.site.core.ws.dto;

import java.time.Instant;
import java.util.List;

public record GameRoom(
        Instant createdAt,
        String roomId,
        List<GameRoomMember> members,
        GameRoomMember host
) implements GameRoomBase { }

File: frontend/src/app/sites/core/ws/dto/GameRoom.ts
import {GameRoomMember} from "./GameRoomMember";
import {GameRoomBase} from "./GameRoomBase";

export interface GameRoom extends GameRoomBase {
    createdAt: string;
    roomId: string;
    members: GameRoomMember[];
    host: GameRoomMember;
}